spec/src/main/asciidoc/circuitbreaker.asciidoc

```diff
@@ -38,7 +38,16 @@ succeed before the circuit can be closed. After the specified number of successf
 before the successThreshold is reached the circuit will transition to open. 
 
 Note that circuit state transitions will reset the Circuit Breaker's records. For example, when the circuit transitions to closed a new
-rolling failure window is created with the configured requestVolumeThreshold and failureRatio. The circuit state will only be assessed when the rolling window reaches the `requestVolumeThreshold`. 
+rolling failure window is created with the configured requestVolumeThreshold and failureRatio. The circuit state will only be assessed when the rolling window reaches the `requestVolumeThreshold`.
+
+When a method returns a result, the following rules are applied to determine whether the result is a success or a failure:
+
+* If the method does not throw a `Throwable`, it is considered a success
+* Otherwise, if the thrown object is assignable to any value in the `skipOn` parameter, is is considered a success
+* Otherwise, if the thrown object is assignable to any value in the `failOn` parameter, it is considered a failure
+* Otherwise it is considered a success
+
+If a method throws a `Throwable` which is not a subclass of either `Error` or `Exception`, non-portable behavior results.
```

Comment:
I personally consider the Javadoc text to be part of the specification, so I'd say we don't have to duplicate this info in the spec text.

Reply:
@Ladicek I think for this kind of info, we need to put on JavaDoc to ensure the developers see this. The reason is that with the javadoc, IDE can display it, which is the point you need the info. Sure, someone might read the spec and they might have forgotten when they code.

Reply:
Personally, I think that custom throwables are sufficiently rare case that we don't have to add the info into Javadoc.

On the other hand, I'd be totally fine with moving most of the spec text to the Javadocs (because, as I said, I see Javadoc as part of the spec); I just don't see the point of duplicating the info.

On a third hand :-), I don't really mind _that much_.

Reply:
I prefer the Javadoc way. I'll vote to add more text to Javadoc. It does not hurt as the Javadoc directly tells the end users what to do.

Reply:
I don't oppose, but we should probably discuss elsewhere (separate issue?) the relationship between the spec text and the javadoc. Some people might see them as separate (spec text is canonical source of truth, while javadoc is end user documentation), while some might see them as one thing. Perhaps this should be even discussed on the MicroProfile Architecture level.

Reply:
Just wondering -- in the spec text, we already specify that throwing custom throwables is non-portable in the _Fault Tolerance Interceptor_ section. Shall we repeat it again for each FT policy?

Reply:
Yea. I'll consider it a service to spec readers :-)

Reply:
And I did that in my PR to add the retryOn/abortOn algorithm to the spec (#453), so we should be good.